src/util/data.js

```diff
@@ -1,70 +1,154 @@
+                console.log(resource.id);
+                client
+                    .delete("QuestionnaireResponse/" + resource.id)
+                    .then((result) => {
+                        consoleLog("Successfully deleted QuestionnaireResponse " + resource.id + " from EHR", types.info);
+                        console.log(result);
+                    })
+                    .catch((e) => {
+                        console.log("Failed to delete QuestionnaireResponse " + resource.id);
+                        console.log(e);
+                    });
+            });
+        })
+        .catch((e) => {
+            console.log("Failed to retrieve list of QuestionnaireResponses");
+            console.log(e);
+        });
+};
+
+const resetPims = ({ pimsUrl }, consoleLog) => (_event) => {
+    let url = new URL(pimsUrl);
+    const resetUrl = url.origin + "/doctorOrders/api/deleteAll";
+    console.log("reset pims: " + resetUrl);
+    
+    fetch(resetUrl, {
+        method: 'DELETE',
+    })
+    .then(response => {
+        console.log("Reset pims: ");
+        console.log(response);
+        consoleLog("Successfully reset pims database", types.info);
+    })
+    .catch(error => {
+        console.log("Reset pims error: ");
+        consoleLog("Server returned error when resetting pims: ", types.error);
+        consoleLog(error.message);
+        console.log(error);
+    });
 }
 
-const headers = {
-    "ehrUrl": {
-        "display": "EHR Server",
-        "value": (env.get('REACT_APP_EHR_SERVER').asString()),
-        "key": "ehrUrl"
+const resetRemsAdmin = ({ cdsUrl } , consoleLog) => (_event) => {
+    let url = new URL(cdsUrl);
+    const resetUrl = url.origin + "/etasu/reset";
+
+    fetch(resetUrl, {
+        method: 'POST',
+    })
+    .then(response => {
+        console.log("Reset rems admin etasu: ");
+        console.log(response);
+        consoleLog("Successfully reset rems admin etasu", types.info);
+    })
+    .catch(error => {
+        console.log("Reset rems admin error: ");
+        consoleLog("Server returned error when resetting rems admin etasu: ", types.error);
+        consoleLog(error.message);
+        console.log(error);
+    });
+}
+
+
+const headerDefinitions = {
+    alternativeTherapy: { 
+        display: "Alternative Therapy Cards Allowed",
+        type: "check"
+    }, 
+    baseUrl: { 
+        display: "Base Server",
+        type: "input"
     },
-    "cdsUrl": {
-        "display": "CRD Server",
-        "value": (env.get('REACT_APP_CDS_SERVICE').asString()),
-        "key":"cdsUrl"
+    cdsUrl: { 
+        display: "REMS Admin",
+        type: "input"
     },
-    "orderSelect": {
-        "display": "Order Select Rest End Point",
-        "value": (env.get('REACT_APP_ORDER_SELECT').asString()),
-        "key":"orderSelect"
+    clearQuestionnaireResponses: { 
+        display: "Clear EHR QuestionnaireResponses",
+        type: "button",
+        reset: clearQuestionnaireResponses
```

Comment:
this concerns me a bit, given that we're kind of limiting ourselves when it comes to functions we can put in the settings.  I'd rather just take the button type out of this abstraction entirely and do those manually inside the settings box.  There's really no need to have them in a list like this since they don't interact with the state.

Reply:
Done as of https://github.com/mcode/request-generator/pull/53/commits/06f848b1febbf1c6dc7c661bbaaafdfe3a4f5fff